unsigned int tomo_db0::IoLoop::r7_epoll_pass<false, true, true>(int)
--- PRE

+++ POST

@@ -12,160 +12,147 @@

 0022 [ 2] mov %esi,%ecx
 0024 [ 5] mov $0x100,%edx
 0029 [ 3] mov %rbx,%rsi
-002c [ 3] xor %r12d,%r12d
-002f [ 6] mov $0x0,%r13d
+002c [ 3] xor %r13d,%r13d
+002f [ 6] mov $0x0,%r12d
 0035 [ 9] mov %fs:0x28,%rax
 003e [ 5] mov %rax,0x28(%rsp)
 0043 [ 2] xor %eax,%eax
 0045 [ 5] call <epoll_wait@plt>
 004a [ 2] test %eax,%eax
-004c [ 4] cmovns %eax,%r12d
-0050 [ 7] mov %r12d,0xc18(%r15)
-0057 [ 6] jle <+0x13f>
-005d [ 7] mov 0x190(%r15),%rax
-0064 [ 3] movslq %r12d,%rdx
-0067 [ 2] xor %ebp,%ebp
-0069 [ 7] lea <rip>(%rip),%r14 # <local-switch-0x130,0x2d0,0x250,0x130,0x130,0x170,0x130,0x130,0x120,0x220,0x130,0x130,0x130,0x130,0x130,0x130,0x130,0x1e0>
-0070 [ 7] add %rdx,0x520(%rax)
-0077 [ 7] lea 0xac8(%r15),%rax
-007e [ 5] mov %rax,0x10(%rsp)
-0083 [11] data16 cs nopw 0x0(%rax,%rax,1)
-008e [ 2] xchg %ax,%ax
-0090 [ 4] mov 0x4(%rbx),%rdx
-0094 [ 3] mov %rdx,%rax
-0097 [ 4] shr $0x38,%rax
-009b [ 2] cmp $0x11,%al
-009d [ 6] ja <+0x130>
-00a3 [ 4] movslq (%r14,%rax,4),%rax
-00a7 [ 3] add %r14,%rax
-00aa [ 3] notrack jmp *%rax
-00ad [ 2] mov (%rbx),%eax
-00af [ 5] test $0x2019,%eax
-00b4 [ 2] je <+0xba>
-00b6 [ 4] movb $0x0,0x29(%rdx)
-00ba [ 2] test $0x1c,%al
-00bc [ 2] je <+0xf0>
-00be [ 4] cmpb $0x0,0x30(%rdx)
-00c2 [ 5] mov %rdx,0x20(%rsp)
-00c7 [ 2] jne <+0xf0>
-00c9 [ 5] lea 0x20(%rsp),%rax
-00ce [ 7] lea 0x198(%r15),%rdi
-00d5 [ 4] movb $0x1,0x30(%rdx)
-00d9 [ 5] mov %rdx,0x18(%rsp)
-00de [ 3] mov %rax,%rsi
-00e1 [ 5] mov %rax,0x8(%rsp)
-00e6 [ 5] call <std::deque<tomo_db0::Client*, std::allocator<tomo_db0::Client*> >::push_back(tomo_db0::Client* const&)>
-00eb [ 5] mov 0x18(%rsp),%rdx
-00f0 [ 3] mov %rdx,%rsi
-00f3 [ 3] mov %r15,%rdi
-00f6 [ 5] mov %rdx,0x8(%rsp)
-00fb [ 5] call <tomo_db0::IoLoop::tls_slot_conn(tomo_db0::Client const*)>
-0100 [ 5] mov 0x8(%rsp),%rdx
-0105 [ 3] test %rax,%rax
-0108 [ 2] je <+0x114>
-010a [ 6] movw $0x0,0x3e(%rax)
-0110 [ 4] movb $0x0,0x29(%rdx)
-0114 [ 3] mov %rdx,%rsi
-0117 [ 3] mov %r15,%rdi
-011a [ 5] call <tomo_db0::IoLoop::mark_active(tomo_db0::Client*)>
-011f [ 1] nop
-0120 [ 3] inc %r13d
-0123 [11] data16 cs nopw 0x0(%rax,%rax,1)
-012e [ 2] xchg %ax,%ax
-0130 [ 2] inc %ebp
-0132 [ 4] add $0xc,%rbx
-0136 [ 3] cmp %r12d,%ebp
-0139 [ 6] jl <+0x90>
-013f [ 5] mov 0x28(%rsp),%rax
-0144 [ 9] sub %fs:0x28,%rax
-014d [ 6] jne <+0x2f4>
-0153 [ 4] add $0x38,%rsp
-0157 [ 3] mov %r13d,%eax
-015a [ 1] pop %rbx
-015b [ 1] pop %rbp
-015c [ 2] pop %r12
-015e [ 2] pop %r13
-0160 [ 2] pop %r14
-0162 [ 2] pop %r15
-0164 [ 1] ret
-0165 [11] data16 cs nopw 0x0(%rax,%rax,1)
-0170 [ 5] lea 0x20(%rsp),%rax
-0175 [ 9] movq $0x0,0x20(%rsp)
-017e [ 5] mov %rax,0x8(%rsp)
-0183 [11] data16 cs nopw 0x0(%rax,%rax,1)
-018e [ 2] xchg %ax,%ax
-0190 [ 7] mov 0xbc8(%r15),%edi
-0197 [ 5] mov 0x8(%rsp),%rsi
-019c [ 5] mov $0x8,%edx
-01a1 [ 5] call <read@plt>
-01a6 [ 4] cmp $0x8,%rax
-01aa [ 2] je <+0x190>
-01ac [ 7] mov 0x190(%r15),%rax
-01b3 [ 2] inc %ebp
-01b5 [ 3] inc %r13d
-01b8 [ 4] add $0xc,%rbx
-01bc [ 7] incq 0x440(%rax)
-01c3 [ 3] cmp %r12d,%ebp
-01c6 [ 6] jl <+0x90>
-01cc [ 5] jmp <+0x13f>
-01d1 [11] data16 cs nopw 0x0(%rax,%rax,1)
-01dc [ 4] nopl 0x0(%rax)
-01e0 [ 7] mov 0x190(%r15),%rax
-01e7 [ 5] mov 0x10(%rsp),%rdx
-01ec [ 7] lea <rip>(%rip),%rdi # <tomo_db0::Server::databases_>
-01f3 [ 2] inc %ebp
-01f5 [ 3] inc %r13d
-01f8 [ 4] add $0xc,%rbx
-01fc [ 2] mov (%rax),%esi
-01fe [ 5] call <tomo_db0::DatabaseMap::consume_wake(unsigned int, tomo_db0::Ring&)>
-0203 [ 3] cmp %r12d,%ebp
-0206 [ 6] jl <+0x90>
-020c [ 5] jmp <+0x13f>
-0211 [11] data16 cs nopw 0x0(%rax,%rax,1)
-021c [ 4] nopl 0x0(%rax)
-0220 [ 5] mov $0x9,%esi
-0225 [ 3] mov %r15,%rdi
-0228 [ 2] inc %ebp
-022a [ 4] add $0xc,%rbx
-022e [ 5] call <unsigned int tomo_db0::IoLoop::r7_epoll_accept<true, true>(tomo_db0::UrKind)>
-0233 [ 3] add %eax,%r13d
-0236 [ 3] cmp %r12d,%ebp
-0239 [ 6] jl <+0x90>
-023f [ 5] jmp <+0x13f>
-0244 [11] data16 cs nopw 0x0(%rax,%rax,1)
-024f [ 1] nop
-0250 [10] movabs $0xffffffffffff,%rax
-025a [ 3] and %rax,%rdx
-025d [ 6] je <+0x130>
-0263 [ 6] mov 0x788(%rdx),%eax
-0269 [ 7] mov 0x190(%r15),%rcx
-0270 [ 2] cmp %eax,(%rcx)
-0272 [ 6] jne <+0x130>
-0278 [ 4] cmpb $0x0,0x33(%rdx)
-027c [ 6] jne <+0x130>
-0282 [ 6] mov 0x784(%rdx),%eax
-0288 [ 3] cmp $0xfffffffe,%eax
-028b [ 6] je <+0x130>
-0291 [ 7] mov 0x288(%r15),%rax
-0298 [ 7] mov 0x290(%r15),%rcx
-029f [ 3] cmp %rcx,%rax
-02a2 [ 2] jne <+0x2bd>
-02a4 [ 5] jmp <+0xad>
-02a9 [ 7] nopl 0x0(%rax)
-02b0 [ 4] add $0x28,%rax
-02b4 [ 3] cmp %rax,%rcx
-02b7 [ 6] je <+0xad>
-02bd [ 3] cmp (%rax),%rdx
-02c0 [ 2] jne <+0x2b0>
-02c2 [ 5] jmp <+0x130>
-02c7 [ 9] nopw 0x0(%rax,%rax,1)
-02d0 [ 5] mov $0x1,%esi
-02d5 [ 3] mov %r15,%rdi
-02d8 [ 2] inc %ebp
-02da [ 4] add $0xc,%rbx
-02de [ 5] call <unsigned int tomo_db0::IoLoop::r7_epoll_accept<true, true>(tomo_db0::UrKind)>
-02e3 [ 3] add %eax,%r13d
-02e6 [ 3] cmp %r12d,%ebp
-02e9 [ 6] jl <+0x90>
-02ef [ 5] jmp <+0x13f>
-02f4 [ 5] call <__stack_chk_fail@plt>
+004c [ 4] cmovns %eax,%r13d
+0050 [ 7] mov %r13d,0xc18(%r15)
+0057 [ 2] jle <+0xbb>
+0059 [ 7] mov 0x190(%r15),%rax
+0060 [ 3] movslq %r13d,%rdx
+0063 [ 2] xor %ebp,%ebp
+0065 [ 7] lea <rip>(%rip),%r14 # <local-switch-0xa0,0x240,0xf0,0xa0,0xa0,0x1e0,0xb0,0xa0,0x1d0,0x290,0xa0,0xa0,0xa0,0xa0,0xa0,0xa0,0xa0,0x260>
+006c [ 7] add %rdx,0x520(%rax)
+0073 [ 7] lea 0xac8(%r15),%rax
+007a [ 5] mov %rax,0x10(%rsp)
+007f [ 1] nop
+0080 [ 4] mov 0x4(%rbx),%rsi
+0084 [ 3] mov %rsi,%rax
+0087 [ 4] shr $0x38,%rax
+008b [ 2] cmp $0x11,%al
+008d [ 2] ja <+0xa0>
+008f [ 4] movslq (%r14,%rax,4),%rax
+0093 [ 3] add %r14,%rax
+0096 [ 3] notrack jmp *%rax
+0099 [ 7] nopl 0x0(%rax)
+00a0 [ 3] mov %r15,%rdi
+00a3 [ 5] call <tomo_db0::IoLoop::fliplisten_epoll(unsigned long)>
+00a8 [ 3] add %eax,%r12d
+00ab [ 5] nopl 0x0(%rax,%rax,1)
+00b0 [ 2] inc %ebp
+00b2 [ 4] add $0xc,%rbx
+00b6 [ 3] cmp %r13d,%ebp
+00b9 [ 2] jl <+0x80>
+00bb [ 5] mov 0x28(%rsp),%rax
+00c0 [ 9] sub %fs:0x28,%rax
+00c9 [ 6] jne <+0x2a5>
+00cf [ 4] add $0x38,%rsp
+00d3 [ 3] mov %r12d,%eax
+00d6 [ 1] pop %rbx
+00d7 [ 1] pop %rbp
+00d8 [ 2] pop %r12
+00da [ 2] pop %r13
+00dc [ 2] pop %r14
+00de [ 2] pop %r15
+00e0 [ 1] ret
+00e1 [11] data16 cs nopw 0x0(%rax,%rax,1)
+00ec [ 4] nopl 0x0(%rax)
+00f0 [10] movabs $0xffffffffffff,%rdx
+00fa [ 3] and %rsi,%rdx
+00fd [ 2] je <+0xb0>
+00ff [ 6] mov 0x788(%rdx),%eax
+0105 [ 7] mov 0x190(%r15),%rcx
+010c [ 2] cmp %eax,(%rcx)
+010e [ 2] jne <+0xb0>
+0110 [ 4] cmpb $0x0,0x33(%rdx)
+0114 [ 2] jne <+0xb0>
+0116 [ 6] mov 0x784(%rdx),%eax
+011c [ 3] cmp $0xfffffffe,%eax
+011f [ 2] je <+0xb0>
+0121 [ 7] mov 0x288(%r15),%rax
+0128 [ 7] mov 0x290(%r15),%rcx
+012f [ 3] cmp %rcx,%rax
+0132 [ 2] je <+0x152>
+0134 [11] data16 cs nopw 0x0(%rax,%rax,1)
+013f [ 1] nop
+0140 [ 3] cmp (%rax),%rdx
+0143 [ 6] je <+0xb0>
+0149 [ 4] add $0x28,%rax
+014d [ 3] cmp %rax,%rcx
+0150 [ 2] jne <+0x140>
+0152 [ 2] mov (%rbx),%eax
+0154 [ 5] test $0x2019,%eax
+0159 [ 2] je <+0x15f>
+015b [ 4] movb $0x0,0x29(%rdx)
+015f [ 2] test $0x1c,%al
+0161 [ 2] je <+0x195>
+0163 [ 4] cmpb $0x0,0x30(%rdx)
+0167 [ 5] mov %rdx,0x20(%rsp)
+016c [ 2] jne <+0x195>
+016e [ 5] lea 0x20(%rsp),%rax
+0173 [ 7] lea 0x198(%r15),%rdi
+017a [ 4] movb $0x1,0x30(%rdx)
+017e [ 5] mov %rdx,0x18(%rsp)
+0183 [ 3] mov %rax,%rsi
+0186 [ 5] mov %rax,0x8(%rsp)
+018b [ 5] call <std::deque<tomo_db0::Client*, std::allocator<tomo_db0::Client*> >::push_back(tomo_db0::Client* const&)>
+0190 [ 5] mov 0x18(%rsp),%rdx
+0195 [ 3] mov %rdx,%rsi
+0198 [ 3] mov %r15,%rdi
+019b [ 5] mov %rdx,0x8(%rsp)
+01a0 [ 5] call <tomo_db0::IoLoop::tls_slot_conn(tomo_db0::Client const*)>
+01a5 [ 5] mov 0x8(%rsp),%rdx
+01aa [ 3] test %rax,%rax
+01ad [ 2] je <+0x1b9>
+01af [ 6] movw $0x0,0x3e(%rax)
+01b5 [ 4] movb $0x0,0x29(%rdx)
+01b9 [ 3] mov %rdx,%rsi
+01bc [ 3] mov %r15,%rdi
+01bf [ 5] call <tomo_db0::IoLoop::mark_active(tomo_db0::Client*)>
+01c4 [11] data16 cs nopw 0x0(%rax,%rax,1)
+01cf [ 1] nop
+01d0 [ 3] inc %r12d
+01d3 [ 5] jmp <+0xb0>
+01d8 [ 8] nopl 0x0(%rax,%rax,1)
+01e0 [ 5] lea 0x20(%rsp),%rax
+01e5 [ 9] movq $0x0,0x20(%rsp)
+01ee [ 5] mov %rax,0x8(%rsp)
+01f3 [11] data16 cs nopw 0x0(%rax,%rax,1)
+01fe [ 2] xchg %ax,%ax
+0200 [ 7] mov 0xbc8(%r15),%edi
+0207 [ 5] mov 0x8(%rsp),%rsi
+020c [ 5] mov $0x8,%edx
+0211 [ 5] call <read@plt>
+0216 [ 4] cmp $0x8,%rax
+021a [ 2] je <+0x200>
+021c [ 7] mov 0x190(%r15),%rax
+0223 [ 3] inc %r12d
+0226 [ 7] incq 0x440(%rax)
+022d [ 5] jmp <+0xb0>
+0232 [11] data16 cs nopw 0x0(%rax,%rax,1)
+023d [ 3] nopl (%rax)
+0240 [ 5] mov $0x1,%esi
+0245 [ 3] mov %r15,%rdi
+0248 [ 5] call <unsigned int tomo_db0::IoLoop::r7_epoll_accept<true, true>(tomo_db0::UrKind)>
+024d [ 3] add %eax,%r12d
+0250 [ 5] jmp <+0xb0>
+0255 [11] data16 cs nopw 0x0(%rax,%rax,1)
+0260 [ 7] mov 0x190(%r15),%rax
+0267 [ 5] mov 0x10(%rsp),%rdx
+026c [ 7] lea <rip>(%rip),%rdi # <tomo_db0::Server::databases_>
+0273 [ 3] inc %r12d
+0276 [ 2] mov (%rax),%esi
+0278 [ 5] call <tomo_db0::DatabaseMap::consume_wake(unsigned int, tomo_db0::Ring&)>
+027d [ 5] jmp <+0xb0>
+0282 [11] data16 cs nopw 0x0(%rax,%rax,1)
+028d [ 3] nopl (%rax)
+0290 [ 5] mov $0x9,%esi
+0295 [ 3] mov %r15,%rdi
+0298 [ 5] call <unsigned int tomo_db0::IoLoop::r7_epoll_accept<true, true>(tomo_db0::UrKind)>
+029d [ 3] add %eax,%r12d
+02a0 [ 5] jmp <+0xb0>
+02a5 [ 5] call <__stack_chk_fail@plt>
